ethereum/contracts/DelegateCashMock.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity >=0.8.10 <0.9.0;
+
+interface MockDelegateCashInterface {
+    function checkDelegateForToken(
+        address delegate,
+        address vault,
+        address contract_,
+        uint256 tokenId
+    ) external view returns (bool);
+
+    function registerDelegate(
+        address delegate,
+        address vault,
+        address contract_,
+        uint256 tokenId
+    ) external;
+}
+
+contract DelegateCashMock is MockDelegateCashInterface {
```

Comment:
so is there one "real" public delegatecash contract out in the wild that we'll use?

Reply:
Yeah. delegate.cash is deployed to the same address on most networks (https://docs.delegate.cash/delegatecash/technical-documentation/delegation-registry/contract-addresses), but we can't add new constants to the contract since it is upgradeable so I think we need to use a setter instead?

Reply:
we could add new consts... I think it's just a matter of if we want to. may as well to avoid a new method? note: consts can change with upgrades since they don't use contract storage.

Reply:
yeah I agree. I was going to do that first but for some reason I thought that we couldn't add a constant to an upgradeable contract. fixed!